src/Frontend/Core.hs

```diff
@@ -454,6 +458,51 @@ ideasFilterQuery = \case
     (Just cat) -> filter ((== Just cat) . view ideaCategory)
     Nothing    -> id
 
+type IdeasSortApi = QueryParam "sortby" SortIdeasBy
+type IdeasSortQuery = Maybe SortIdeasBy
+
+data SortIdeasBy = SortIdeasByAge | SortIdeasBySupport
+  deriving (Eq, Ord, Show, Read, Enum, Bounded)
+
+instance FromHttpApiData SortIdeasBy where
+    parseUrlPiece = \case
+        "age" -> Right SortIdeasByAge
+        "sup" -> Right SortIdeasBySupport
+        _     -> Left "no parse"
+
+instance ToHttpApiData SortIdeasBy where
+    toUrlPiece = \case
+        SortIdeasByAge     -> "age"
+        SortIdeasBySupport -> "sup"
+
+ideasSortQuery :: IdeasSortQuery -> [Idea] -> [Idea]
+ideasSortQuery = f . fromMaybe minBound
+  where
+    f SortIdeasByAge     = age
+    f SortIdeasBySupport = sup . age
+
+    age = sortOn $ ideaLikes . to length
+    sup = sortOn createdAt
+
+type IdeasQuery = (IdeasFilterQuery, IdeasSortQuery)
+
+ideasRunQuery :: IdeasQuery -> [Idea] -> [Idea]
+ideasRunQuery (f, s) = ideasSortQuery s . ideasFilterQuery f
+
+listIdeasWithQuery :: IdeaLocation -> IdeasQuery -> URL
+listIdeasWithQuery loc (qf, qs) = (absoluteUriPath . relPath . P.listIdeas $ loc)
+                               <> renderBoth (catMaybes [renderFilter <$> qf, renderSort <$> qs])
+  where
+    renderFilter :: Category -> ST
+    renderFilter v = "category=" <> toUrlPiece v
```

Comment:
Can't we reuse the fact that we already declared "category" as being the name of this query param? I can't find anything obvious.